shared/fs/container.js

```diff
@@ -11,62 +11,51 @@ type OwnProps = {
   progress: Types.ProgressType,
-  sortSetting: Types.SortSetting,
-  username?: string,
-  pathItems: I.Map<Types.Path, Types.PathItem>,
 }
 
-type _sortSettingSetter = (setting: Types._SortSetting) => void
 type DispatchProps = {
   loadFolderList: (path: Types.Path) => void,
-  _sortSettingChangeFactory: (path: Types.Path) => _sortSettingSetter,
 }
 
 const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
   const path = Types.stringToPath(routeProps.get('path', Constants.defaultPath))
-  const sortSetting = state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort')
   const itemDetail = state.fs.pathItems.get(path)
-  const itemNames =
-    itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
-  const progress: Types.ProgressType = itemDetail ? itemDetail.progress : 'pending'
-  const username = state.config.username || undefined
-  const pathItems = state.fs.pathItems
-  return {itemNames, path, progress, sortSetting, username, pathItems}
+  return {
+    _itemNames: itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List(),
+    _username: state.config.username || undefined,
+    _pathItems: state.fs.pathItems,
+    _sortSetting: state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort'),
+    path,
+    progress: itemDetail ? itemDetail.progress : 'pending',
+  }
 }
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
-  _sortSettingChangeFactory: (path: Types.Path) => (setting: Types.SortSetting) =>
-    dispatch(FsGen.createSortSetting({path: path, sortSetting: Constants.makeSortSetting(setting)})),
 })
 
-const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps) => {
-  const setSortSetting = dispatchProps._sortSettingChangeFactory(stateProps.path)
-  return {
-    items: Constants.sortPathItems(
-      stateProps.itemNames.map(name => Types.pathConcat(stateProps.path, name)).map(
-        p => stateProps.pathItems.get(p, Constants.makeUnknownPathItem()) // provide an unknown default to make flow happy
-      ),
-      stateProps.sortSetting,
-      Types.pathIsNonTeamTLFList(stateProps.path) ? stateProps.username : undefined
-    )
-      .map(({name}) => Types.pathConcat(stateProps.path, name))
-      .toArray(),
-    progress: stateProps.progress,
-    path: stateProps.path,
-    sortSetting: stateProps.sortSetting,
-
-    setSortByNameAsc: () => setSortSetting({sortBy: 'name', sortOrder: 'asc'}),
-    setSortByNameDesc: () => setSortSetting({sortBy: 'name', sortOrder: 'desc'}),
-    setSortByTimeAsc: () => setSortSetting({sortBy: 'time', sortOrder: 'asc'}),
-    setSortByTimeDesc: () => setSortSetting({sortBy: 'time', sortOrder: 'desc'}),
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps) => ({
+  items: Constants.sortPathItems(
```

Comment:
lets only inline if its very simple. if you have to do a bunch of transforms lets calculate this first then write to a const then put that into the return value
akak
```
const items = ...
return {
  items,
...
}
```